File: Dijkstra.js
///////////////////////////////////////graph Data
const vertexNum = 6;
const edgeNum = 9;
const graph = [
    [0, 1, 7],
    [0, 2, 9],
    [0, 5, 14],
    [1, 2, 10],
    [1, 3, 15],
    [2, 3, 11],
    [2, 5, 2],
    [3, 4, 6],
    [4, 5, 9],
];
/////////////////////////////////////////////////

//////////////////////////////////Array & Variable
let neighbors = new Array(vertexNum);
for(let i = 0 ; i < neighbors.length;i++){
    neighbors[i] = new Array(vertexNum).fill(-1)
}
let dist = new Array(vertexNum).fill(1000000000);
let prev = new Array(vertexNum).fill(-2);
let q = new Array(vertexNum).fill(-1);

let origin = 0; //starter point
const dest = 4; //destination point
//////////////////////////////////////////////////

/////////////////////////////////////////Functions
function setNeighbors() {
    let counter = new Array(vertexNum).fill(0);
    for (let i = 0; i < edgeNum; i++){
        neighbors[graph[i][0]][counter[graph[i][0]]] = graph[i][1];
        counter[graph[i][0]]++;
        neighbors[graph[i][1]][counter[graph[i][1]]] = graph[i][0];
        counter[graph[i][1]]++;
    }
}

function findEdge(m,d)
{
    for (let i = 0; i < edgeNum; i++){
        if ((graph[i][0] == m || graph[i][1] == m) && (graph[i][1] == d || graph[i][0] == d)){
            return graph[i][2];
        }
    }
    return -1;
}

function checkNeighbors()
{
    for (let i = 0; neighbors[origin][i] != -1; i++){
        let n = neighbors[origin][i];
        if (dist[n] > dist[origin] + findEdge(origin, n)){
            dist[n] = dist[origin] + findEdge(origin, n);
            prev[n] = origin;
        }
    }
    q[origin] = 1;
}
function nextVertexSelector()
{
    let min = 1000000000;
    let res = -1;
    for (let i = 0; i < vertexNum; i++){
        if (dist[i] < min && q[i] == -1){
            min = dist[i];
            res = i;
        }
    }
    return res;
}

function checkNull(){
    for (let i = 0; i < vertexNum; i++){
        if (q[i] == -1)
            return true;
    }
    return false;
}

function printRoad()
{
    let reRoad = [];
    let pointer = dest;
    let i;
    let printExp = ""
    for (i = 0; pointer != -1; i++){
        reRoad.push(pointer);
        pointer = prev[pointer];
    }
    for (i--; i >= 0; i--){
        printExp +=  i != 0 ? `${reRoad[i]} => ` : reRoad[i]
    }
    console.log(printExp); 
}

function validation(){
    for(let i = 0;i < edgeNum;i++){
        if(graph[i][0] == graph[i][1] || graph[i][2] <= 0)
            return false;
    }
    return true;
}
////////////////////////////////////////////////////

////////////////////////////////////////////////Main
if(validation()){
    setNeighbors();
    dist[origin] = 0;
    prev[origin] = -1;
    while (checkNull()){
        checkNeighbors();
        origin = nextVertexSelector();
    }
    printRoad();
}
else
    console.log("Your graph its wrong");

File: Dijkstra.cpp
#include <iostream>
#include <string.h>

using namespace std;
///////////////////////////////////////Graph Data
const int Vertex_num = 6;
const int Edge_num = 9;
int Graph[Edge_num][3] = {
    {0, 1, 7},
    {0, 2, 9},
    {0, 5, 14},
    {1, 2, 10},
    {1, 3, 15},
    {2, 3, 11},
    {2, 5, 2},
    {3, 4, 6},
    {4, 5, 9},
};
/////////////////////////////////////////////////

//////////////////////////////////Array & Variable
int Neighbors[Vertex_num][Vertex_num];
int Dist[Vertex_num];
int Prev[Vertex_num];
int q[Vertex_num];
int origin, dest;
//////////////////////////////////////////////////

/////////////////////////////////////////Functions
void SetNeighbors()
{
    int counter[Vertex_num] = {0};
    for (int i = 0; i < Vertex_num; i++)
    {
        for (int j = 0; j < Vertex_num; j++)
        {
            Neighbors[i][j] = -1;
        }
    }
    for (int i = 0; i < Edge_num; i++)
    {
        Neighbors[Graph[i][0]][counter[Graph[i][0]]] = Graph[i][1];
        counter[Graph[i][0]]++;
        Neighbors[Graph[i][1]][counter[Graph[i][1]]] = Graph[i][0];
        counter[Graph[i][1]]++;
    }
}

void PreSet()
{
    for (int i = 0; i < Vertex_num; i++)
    {
        Dist[i] = 1000000000;
        Prev[i] = -2;
        q[i] = -1;
    }
    Dist[origin] = 0;
    Prev[origin] = -1;
}

int FindEdge(int m, int d)
{
    for (int i = 0; i < Edge_num; i++)
    {
        if (Graph[i][0] == m || Graph[i][1] == m)
        {
            if (Graph[i][1] == d || Graph[i][0] == d)
                return Graph[i][2];
        }
    }
    return -1;
}

void CheckNeighbors()
{
    for (int i = 0; Neighbors[origin][i] != -1; i++)
    {
        int n = Neighbors[origin][i];
        if (Dist[n] > Dist[origin] + FindEdge(origin, n))
        {
            Dist[n] = Dist[origin] + FindEdge(origin, n);
            Prev[n] = origin;
        }
    }
    q[origin] = 1;
}
int NextVertexSelector()
{
    int min = 1000000000;
    int res = -1;
    for (int i = 0; i < Vertex_num; i++)
    {
        if (Dist[i] < min && q[i] == -1)
        {
            min = Dist[i];
            res = i;
        }
    }
    return res;
}

bool CheckNull()
{
    for (int i = 0; i < Vertex_num; i++)
    {
        if (q[i] == -1)
            return true;
    }
    return false;
}

void MakeRoad()
{
    int ReRoad[Vertex_num];
    int pointer = dest;
    int i = 0;
    for (i = 0; pointer != -1; i++)
    {
        ReRoad[i] = pointer;
        pointer = Prev[pointer];
    }
    for (i--; i >= 0; i--)
    {
        if (i != 0)
            cout << ReRoad[i] << " => ";
        else
            cout << ReRoad[i];
    }
}
bool Validation(){
    for(int i = 0;i < Edge_num;i++){
        if(Graph[i][0] == Graph[i][1] || Graph[i][2] <= 0)
            return false;
    }
    return true;
}
////////////////////////////////////////////////////

////////////////////////////////////////////////Main
int main()
{
    if(Validation()){
        cout << "Enter Your Starter Point : ";
        cin >> origin;
        cout << "Enter Your final Point : ";
        cin >> dest;
        SetNeighbors();
        PreSet();
        while (CheckNull())
        {
            CheckNeighbors();
            origin = NextVertexSelector();
        }
        MakeRoad();
    }
    else
        cout<<"Your Graph its wrong";
}